src/plugins/vis_augmenter/public/saved_augment_vis/utils/helpers.ts

```diff
@@ -3,14 +3,47 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { getSavedAugmentVisLoader } from '../../services';
+import { get } from 'lodash';
+import { getSavedAugmentVisLoader, getUISettings } from '../../services';
 import { ISavedAugmentVis } from '../types';
+import {
+  PLUGIN_AUGMENTATION_ENABLE_SETTING,
+  PLUGIN_AUGMENTATION_MAX_OBJECTS_SETTING,
+} from '../../../common/constants';
 
 /**
  * Create an augment vis saved object given an object that
  * implements the ISavedAugmentVis interface
  */
 export const createAugmentVisSavedObject = async (AugmentVis: ISavedAugmentVis): Promise<any> => {
   const loader = getSavedAugmentVisLoader();
+  const config = getUISettings();
+
+  const isAugmentationEnabled = config.get(PLUGIN_AUGMENTATION_ENABLE_SETTING);
+  if (!isAugmentationEnabled) {
+    throw new Error(
+      'Visualization augmentation is disabled, please enable visualization:enablePluginAugmentation.'
+    );
+  }
+  const maxAssociatedCount = config.get(PLUGIN_AUGMENTATION_MAX_OBJECTS_SETTING);
+
+  await loader.findAll().then(async (resp) => {
+    if (resp !== undefined) {
+      const savedAugmentObjects = get(resp, 'hits', []);
+      // gets all the saved object for this visualization
+      const savedObjectsForThisVisualization = savedAugmentObjects.filter(
+        (savedObj) => get(savedObj, 'visId', '') === AugmentVis.visId
+      );
+
+      if (maxAssociatedCount <= savedObjectsForThisVisualization.length) {
+        throw new Error(
+          `Cannot associate the plugin resource to the visualization due to the limit of the max
+          amount of associated plugin resources (${maxAssociatedCount}) with
+          ${savedObjectsForThisVisualization.length} associated to the visualization`
```

Comment:
Do we want this error message to indicate that `visualization:enablePluginAugmentation.maxPluginObjects` is actually a user-configurable setting?